gwpy/cli/transferfunction.py

```diff
@@ -50,39 +61,200 @@ def arg_channels(cls, parser):
                                          'others will be compared')
         return group
 
+    @classmethod
+    def arg_yaxis(cls, parser):
+        group = cls._arg_axis('ymag', parser, scale='log')
+        group.add_argument('--plot-dB', action='store_true',
+                           help='Plot transfer function in dB')
+        group = cls._arg_axis('yphase', parser, scale='linear')
+
+        return group
+
+    def get_title(self):
+        gps = self.start_list[0]
+        utc = Time(gps, format='gps', scale='utc').iso
+        tstr = f'{utc} | {gps} ({self.duration})'
+
+        fftstr = f'fftlength={self.args.secpfft}, overlap={self.args.overlap}'
+
+        return ', '.join([tstr, fftstr])
+
     def _finalize_arguments(self, args):
-        if args.yscale is None:
-            args.yscale = 'linear'
-        if args.yscale == 'linear':
-            if not args.ymin:
-                args.ymin = 0
-            if not args.ymax:
-                args.ymax = 1.05
+        if args.ymagscale is None:
+            args.ymagscale = 'log'
+        if args.plot_dB:
+            args.ymagscale = 'linear'
+
         return super()._finalize_arguments(args)
 
     def get_ylabel(self):
         """Text for y-axis label
         """
-        return 'Transfer function'
+        ylabelstr = ''
+        if self.subplot == 0:
+            ylabelstr = 'Magnitude'
+            if self.plot_dB:
+                ylabelstr += ' [dB]'
+        if self.subplot == 1:
+            ylabelstr = 'Phase [deg.]'
+
+        return ylabelstr
 
     def get_suptitle(self):
         """Start of default super title, first channel is appended to it
         """
-        return 'Transfer function: {0}'.format(self.ref_chan)
+        return f"Transfer function: {self.test_chan}/{self.ref_chan}"
+
+    def set_axes_properties(self):
+
+        for subplot in [0, 1]:
+            self.subplot = subplot
+            self.scale_axes_from_data()
+            self.set_xaxis_properties()
+            self.set_yaxis_properties()
+
+    def _set_axis_properties(self, axis):
+        """Generic method to set properties for X/Y axis
+        on a specific subplot
+        """
+        def _get(param):
+            if axis.lower().startswith('y'):
+                ret = getattr(self.plot.axes[self.subplot], f'get_y{param}')()
+            else:
+                ret = getattr(self.plot.axes[self.subplot], f'get_x{param}')()
```

Comment:
I think this can be simplified:

```suggestion
            ret = getattr(self.plot.axes[self.subplot], f'get_{axis[0].lower()}{param}')()
```

Reply:
There a number of similar cases below. I think overloading the `axis` property for `ymag` and `yphase` is a bit odd, can't really verbalise why.